homeassistant/components/intellidrive/cover.py

```diff
@@ -0,0 +1,153 @@
+"""The cover entity of reisinger intellidrive."""
+
+import logging
+from typing import Any, cast
+
+from homeassistant.components.cover import (
+    CoverDeviceClass,
+    CoverEntity,
+    CoverEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_TOKEN,
+    STATE_CLOSED,
+    STATE_CLOSING,
+    STATE_OPEN,
+    STATE_OPENING,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN, STATUSDICT_OPENSTATE, STATUSDICT_SERIALNO
+from .coordinator import ReisingerCoordinator
+from .entity import IntelliDriveEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+STATES_MAP = {0: STATE_CLOSED, 1: STATE_OPEN}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Intellidrive covers."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        [
+            SlidingDoorCoverEntity(
+                hass,
+                coordinator,
+                str(entry.data.get(CONF_HOST)),
+                str(entry.data.get(CONF_TOKEN)),
```

Comment:
again shouldn't need to do str()

Reply:
Yes that might be correct, but the SlidingDoorCoverEntity expects a string and not an Optinal[any], so I have to do the cast

Reply:
`str(foo)` does not cast, it calls `foo.__str__`

There are two problems here:
- entry.data is not typed, meaning the dict is assumed to be a `dict[Any, Any]`
- you're incorrectly using `dict.get` for items which you know are in the dict any, meaning what you get is assumed to be `Any | None`

you can do like this:
```py
entry.data[CONF_TOKEN],
```

Or even clearer:
```py
token: str = entry.data[CONF_TOKEN]
```

Reply:
Ok I updated this for the CONF_HOST, but the Token is optional so I have to keep the str() in that case